source/contributed/typescript-and-screeps-1.md

```diff
@@ -0,0 +1,207 @@
+
+Lets install the typings for Screeps, Lodash (screeps is using v3.10.1 at the time of this article), and Node (v6.10.3 at the time of this article).  This way the TypeScript compiler can know about these libraries, even though they aren't actually present in your code
+
+```bash
+# (don't worry, global in this context means "ambient" [ see below ]
+# it's not actually installing the typings globally on your computer)
+# Install Screeps Typings
+/my_proj$ typings install --save --global github:screepers/Screeps-Typescript-Declarations/dist/screeps.d.ts 
+# Install Node Typings from definately typed repo for version 6
+/my_proj$ typings install --save --global "dt~node@6.0.0"
+# Install Lodash Typings from definately typed repo for version 3.10
+/my_proj$ typings install --save --global "dt~lodash@3.10.0"
+```
+
+#### Newbie Gotcha: Ambient Declarations vs. Module Declarations
+
+As you may have guessed, typing is a key feature of TypeScript.  However, a common "gotcha" for beginners is the difference between "ambient" (also called "global") typing declarations, and "module" typing declarations.
+
+Above, we installed the typings as "global".  This means the declarations inside are just assumed to be available everywhere in your project, without having to `import` or `require` them.
+
+For example, `Creep` is ambient, because we can just call `Creep.moveTo(...)` without having to first call `import { Creep } from "some-module";`.
+
+Your own code will largely use "module" declarations.  That is, the declaration and type information won't be available in your files, unless you `import` it first.
+
+TypeScript will automatically infer which type you are using in the file.  If your file contains the `export` keyword anywhere, TS will automatically treat the file as a Module.  If not, it will treat the file as ambient (AKA global).
+
+A common mistake is to accidently include `export` somewhere in a file that is meant to be global, and suddenly have typescript complain about being unable to find your definitions because they suddenly switched from "ambient" to "module".
+
+### Compiling with webpack
+
+So you have some code.  Now what?
+
+There are a lot of ways to compile typescript for export (the built in compiler, grunt, gulp, webpack, etc, etc).  Rather than filling this guide with more TS-specific information, this section will merely serve as a guide to point you in the right direction.  For a full-featured setup, see the **screeps-typescript-starter** below.
+
+To start with, at a minimum, you will need [webpack](https://webpack.js.org/guides/webpack-and-typescript/) and either [ts-loader](https://github.com/TypeStrong/ts-loader/) or [awesome-typescript-loader](https://github.com/s-panferov/awesome-typescript-loader)
+(loaders integrate typescript with webpack).  This guide recommends **awesome-typescript-loader**:
+
+```bash
+/my_proj$ npm install --save-dev webpack awesome-typescript-loader
```

Comment:
Seems weird to me that you install webpack again, after you installed it globally before.